posthog/migrations/0307_add_indirect_person_override_constraints.py

```diff
@@ -0,0 +1,96 @@
+# Generated by Django 3.2.16 on 2023-03-01 17:03
+
+import django.contrib.postgres.constraints
+import django.db.models.deletion
+import django.db.models.expressions
+from django.db import migrations, models
+
+# The previous migration attempted to add a constraint to the personoverride
+# table. We want to remove that constraint, as the ForeignKey replaces this.
+# The function was never used, but to make migrations work we need these two
+DROP_FUNCTION_FOR_CONSTRAINT_SQL = "DROP FUNCTION is_override_person_not_used_as_old_person"
+CREATE_FUNCTION_FOR_CONSTRAINT_SQL = f"""
+CREATE OR REPLACE FUNCTION is_override_person_not_used_as_old_person(team_id bigint, override_person_id uuid, old_person_id uuid)
+RETURNS BOOLEAN AS $$
+  SELECT false;
+$$ LANGUAGE SQL;
+"""
+
+DROP_FUNCTION_FOR_CONSTRAINT_SQL = "DROP FUNCTION is_override_person_not_used_as_old_person"
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0306_featureflag_dashboard"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="PersonOverrideHelper",
+            fields=[
+                ("id", models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name="ID")),
+                ("uuid", models.UUIDField()),
+            ],
+        ),
+        migrations.RemoveConstraint(
+            model_name="personoverride",
+            name="unique override per old_person_id",
+        ),
+        migrations.RemoveConstraint(
+            model_name="personoverride",
+            name="old_person_id_is_not_override_person_id",
+        ),
+        migrations.RunSQL(DROP_FUNCTION_FOR_CONSTRAINT_SQL, CREATE_FUNCTION_FOR_CONSTRAINT_SQL),
+        migrations.AddField(
+            model_name="personoverridehelper",
+            name="team",
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to="posthog.team"),
```

Comment:
You're ok with adding these as ForeignKey? Might make things easier to play with/less risky with side-effects of e.g. deleting a team if it was a weaker link.

Reply:
Don't feel super strongly about this constraint, so I'm fine with dropping it. The same one exists on `personoverrides` for now anyways, so we'll catch any unknown teams, but that could also be removed eventually.